src/gwc/src/main/java/org/geoserver/gwc/layer/GeoServerTileLayer.java

```diff
@@ -1522,4 +1534,99 @@ private static String baseUrl() {
         // no HTTP request is in progress
         return null;
     }
+
+    @Override
+    public boolean supportsTileJSON() {
+        return getGridSubsetForSRS(SRS.getEPSG3857()) != null
+                || getGridSubsetForSRS(SRS.getEPSG900913()) != null;
+    }
+
+    @Override
+    public TileJSON getTileJSON() {
+        TileJSON tileJSON = new TileJSON();
+        tileJSON.setName(getName());
+        LayerMetaInformation metaInformation = getMetaInformation();
+        if (metaInformation != null) {
+            tileJSON.setDescription(metaInformation.getDescription());
+        }
+        BoundingBox wgs84Bounds = getBounds(SRS.getEPSG4326());
+        PublishedInfo publishedInfo = getPublishedInfo();
+        PublishedType type = publishedInfo.getType();
+        if (type == PublishedType.VECTOR) {
```

Comment:
Layer groups are marked as PublishedType.GROUP I believe. do we have a test for the group case?

Reply:
Right, I have added code dealing with layerGroup but I didn't modified the condition triggering that code.
Fixing it...